src/renderer/components/SelectAccount.tsx

```diff
@@ -0,0 +1,388 @@
+import { Option as SelectOption } from "@ledgerhq/react-ui/components/form/SelectInput/Option";
+import { Account, AccountLike } from "@ledgerhq/live-common/lib/types";
+import {
+  flattenAccounts,
+  getAccountCurrency,
+  getAccountName,
+  getAccountUnit,
+  listSubAccounts,
+} from "@ledgerhq/live-common/lib/account";
+
+import { Trans, useTranslation } from "react-i18next";
+import { components, createFilter, OptionProps, SingleValueProps } from "react-select";
+import { connect, useDispatch } from "react-redux";
+import { createStructuredSelector } from "reselect";
+import styled, { useTheme } from "styled-components";
+
+import CryptoCurrencyIcon from "./CryptoCurrencyIcon";
+import FormattedVal from "./FormattedVal";
+import AccountTagDerivationMode from "./AccountTagDerivationMode";
+import { shallowAccountsSelector } from "../reducers/accounts";
+import { openModal } from "../actions/modals";
+import Plus from "../icons/Plus";
+
+const IndentLine = styled.div`
+  height: 36px;
+  width: 1px;
+  background: ${p => p.theme.colors.neutral.c70};
+  margin: 0 8px;
+`;
+
+type SelectOption = {
+  matched: boolean;
+  account: AccountLike;
+};
+
+type AccountOptionProps = {
+  account: AccountLike;
+  isSelectedValue?: boolean;
+  hideDerivationTag?: boolean;
+};
+
+export const AccountOption = ({
+  account,
+  isSelectedValue = false,
+  hideDerivationTag = false,
+}: AccountOptionProps) => {
+  const currency = getAccountCurrency(account);
+  const unit = getAccountUnit(account);
+  const name = getAccountName(account);
+  const nested = ["TokenAccount", "ChildAccount"].includes(account.type);
+  const balance =
+    account.type !== "ChildAccount" && account.spendableBalance
+      ? account.spendableBalance
+      : account.balance;
+
+  return (
+    <Flex flexGrow={1} alignItems="center" columnGap={4} height={48}>
+      {!isSelectedValue && nested ? <IndentLine /> : null}
+      {!isSelectedValue && (
+        <Text color="neutral.c100">
+          <CryptoCurrencyIcon currency={currency} size={16} />
+        </Text>
+      )}
+      <Text flex={1} variant="body" fontWeight="medium" color="inherit" fontSize={4}>
+        {name}
+      </Text>
+      {!hideDerivationTag && <AccountTagDerivationMode account={account} />}
+      {!isSelectedValue && unit && balance && (
+        <Text color="neutral.c100" mr={4}>
+          <FormattedVal
+            color="palette.text.shade40"
+            variant="body"
+            fontWeight="medium"
+            fontSize={3}
+            val={balance}
+            unit={unit}
+            showCode
+          />
+        </Text>
+      )}
+    </Flex>
+  );
+};
+
+const defaultRenderOption = (props: OptionProps<SelectOption, false>) => {
+  return (
+    <SelectOption {...props} render={({ data }) => <AccountOption account={data?.account} />} />
+  );
+};
+
+const defaultRenderValue = (props: SingleValueProps<SelectOption>) => {
+  const propsValue = props.getValue();
```

Comment:
I think (but I might be wrong) that `getValue()` will always return an array, so it would be best to check if `props.getValue()[0]` is undefined instead.

<img width="275" alt="Capture d’écran 2021-12-22 à 12 22 28" src="https://user-images.githubusercontent.com/3428394/147085451-fddebb39-0076-4574-8e0d-8fe6c9cb5736.png">

Reply:
Done ✅